include/swift/Driver/Job.h

```diff
@@ -230,9 +230,20 @@ class CommandOutput {
 class Job {
 public:
   enum class Condition {
+    // There was no information about the previous build (i.e., an input map),
+    // or the map marked this Job as dirty or needing a cascading build.
+    // Be maximally conservative with dependencies.
     Always,
+    // The input changed, or this job was scheduled as non-cascading in the last
+    // build
+    // but didn't get to run.
+    // The scheduled-but-didn't-run condition is detected when the job was not
+    // dirty but its primary output was missing.
```

Comment:
This isn't the only such case, so I'm not sure why you're calling it out here.

Reply:
Seemed to be worthwhile for the sake of clarity. I can delete it if you prefer. (Not exactly sure what the antecedent for "this" was in your comment.)

Reply:
I keep doing that. "the job is not dirty but the primary output is missing" is one case where RunWithoutCascading can happen, but it's not the only one.

Reply:
Thanks! I get it and removed that sentence.